.github/workflows/build-and-test-cuda.yaml

```diff
@@ -0,0 +1,33 @@
+name: build-and-test-cuda
+
+on:
+  pull_request:
+    types: [opened, labeled, unlabeled, synchronize]
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-test-cuda:
+    runs-on: [self-hosted, linux, X64]
+    container: ${{ matrix.container }}${{ matrix.container-suffix }}
+    strategy:
+      fail-fast: false
+      matrix:
+        rosdistro:
+          - humble
+        container-suffix:
+          - -cuda
```

Comment:
NOTE: 
We assume that we should be able to catch most of build failures in build-and-test-differential so we will only run against cuda here.